wolfssl/wolfcrypt/types.h

```diff
@@ -422,25 +422,25 @@ typedef struct w64wrapper {
             #include <wolfssl/wolfcrypt/port/intel/quickassist_mem.h>
             #undef USE_WOLFSSL_MEMORY
             #ifdef WOLFSSL_DEBUG_MEMORY
-                #define XMALLOC(s, h, t)     IntelQaMalloc((s), (h), (t), __func__, __LINE__)
-                #define XFREE(p, h, t)       IntelQaFree((p), (h), (t), __func__, __LINE__)
-                #define XREALLOC(p, n, h, t) IntelQaRealloc((p), (n), (h), (t), __func__, __LINE__)
+                #define XMALLOC(s, h, t)     IntelQaMalloc(s, h, t, __func__, __LINE__)
```

Comment:
Please reconsider if you really need to remove the `()` in all these cases. Seems odd that it would cause a C89 issue.

Reply:
they're frivolous parentheses, confusing in the sense of "why are they doing that?  they can't ever do anything."

in macro definitions, the bugprone-parentheses class of problem only happens when macro args aren't delineated by commas and parentheses when used in the macro definition, because they're already delineated by parentheses and commas in the argument list.

so the extra parens in those macro definition are just visual noise.

Reply:
I think we should keep the () in all of these cases if it is not throwing a warning / error. Seem to remember a case like this on the coldfire with an old old compiler and having issues with it. Here is an ugly example of macro no no's that this could help sort out.

```
#include <stdio.h>
#define test(a, b) test_2((a), b)
#define X 3
#define Y 3,2

void test_2(int a, int b)
{
    printf("a = %d\n", a);
}

int main()
{
    test(Y, X);
    return 0;
}
```

Without the extra parentheses it would have build issues. The seed idea for the test case on macro parentheses was from the gcc manual (https://gcc.gnu.org/onlinedocs/cpp/Argument-Prescan.html#Argument-Prescan).